### PR TITLE
Add support for ExtraPartitionSize to address issue #676

### PR DESCRIPTION
**Pull request by xphyr**
# Background / Context

This PR addresses the inability to reserve some space on the install device for other partitions.  By allowing for additional partitions on the install disk we can create partitions for use by other processes, such as the LVM Operator.

# Issue / Requirement / Reason for change

This PR addresses issue #676 

# Solution / Feature Overview

I have added one extra configuration option "ExtraPartitionSize" which allows you to define the size of the varlibcontainers partition that is created. It is a optional configuration. If you do not define this option, the lca-cli works exactly the same as it did prior to this code change. This means that all existing configuration files will work with no changes.

# Implementation Details

This change takes advantage of the existing "sgdisk" code and appends the required additional value of ExtraPartitionSize to the "new" command, allowing us to specify a total size. By defaulting "ExtraPartitionSize" to "0" the any previous config files that did not specify this value will work without any modification. 

# Other Information

Given that MCO can not be used to configure additional partitions the lca-cli utility should be updated to support the creation of additional partitions as part of the image application process. In theory this can be handled by adding a "post.sh" with the additional commands, but it would be easier if it was built into the utility itself.

### CODE REVIEW

File: lca-cli/ibi-preparation/ibipreparation_test.go [hunk 1]
```diff
@@ -2,11 +2,12 @@ package ibi_preparation
 
 import (
 	"fmt"
-	"github.com/openshift-kni/lifecycle-agent/api/ibiconfig"
 	"os"
 	"path"
 	"testing"
 
+	"github.com/openshift-kni/lifecycle-agent/api/ibiconfig"
```
Comment (mresvanis) [inline lca-cli/ibi-preparation/ibipreparation_test.go:9]: There seems to be an issue with this change in the CI, specifically I noticed the following [here](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift-kni_lifecycle-agent/677/pull-ci-openshift-kni-lifecycle-agent-main-ci-job/1854191739665911808):

```shell
go vet ./...
# github.com/openshift-kni/lifecycle-agent/lca-cli/ibi-preparation
vet: lca-cli/ibi-preparation/ibipreparation_test.go:17:2: ibiconfig redeclared in this block
```

I didn't look into the cause of this, but I think if we moved this line to line 16 in the same file (i.e. before the other lifecycle-agent import) , it should resolved the CI issue. @xphyr could you perhaps give it a try?

**Comment by tsorya** [comment]
we found another workaround for it @xphyr maybe we can close it?

**Comment by xphyr** [comment]
> we found another workaround for it @xphyr maybe we can close it?

Yes, I think that makes sense. Thanks for addressing this request.

**Pull request closed without merging**